src/utilities/sequence-editor/sequence-linter.ts

```diff
@@ -1212,79 +1212,56 @@ function validateAndLintArguments(
  * @returns A Diagnostic object representing the validation error, or undefined if there is no error.
  */
 function validateCommandStructure(
   stemNode: SyntaxNode,
   argsNode: SyntaxNode[] | null,
-  exactArgSize: number,
+  expectedArgSize: number,
   addDefault: (view: any) => any,
 ): Diagnostic | undefined {
-  if (arguments.length > 0) {
-    if (!argsNode || argsNode.length === 0) {
-      return {
-        actions: [],
-        from: stemNode.from,
-        message: `The stem is missing arguments.`,
-        severity: 'error',
-        to: stemNode.to,
-      };
-    }
-    if (argsNode.length > exactArgSize) {
-      const extraArgs = argsNode.slice(exactArgSize);
-      const { from, to } = getFromAndTo(extraArgs);
-      return {
-        actions: [
-          {
-            apply(view, from, to) {
-              view.dispatch({ changes: { from, to } });
-            },
-            name: `Remove ${extraArgs.length} extra argument${extraArgs.length > 1 ? 's' : ''}`,
-          },
-        ],
-        from,
-        message: `Extra arguments, definition has ${exactArgSize}, but ${argsNode.length} are present`,
-        severity: 'error',
-        to,
-      };
-    }
-    if (argsNode.length < exactArgSize) {
-      const { from, to } = getFromAndTo(argsNode);
-      const pluralS = exactArgSize > argsNode.length + 1 ? 's' : '';
-      return {
-        actions: [
-          {
-            apply(view) {
-              addDefault(view);
-            },
-            name: `Add default missing argument${pluralS}`,
-          },
-        ],
-        from,
-        message: `Missing argument${pluralS}, definition has ${argsNode.length}, but ${exactArgSize} are present`,
-        severity: 'error',
-        to,
-      };
-    }
-  } else if (argsNode && argsNode.length > 0) {
-    const { from, to } = getFromAndTo(argsNode);
+  if ((!argsNode || argsNode.length === 0) && expectedArgSize === 0) {
+    return undefined;
+  }
+  if (argsNode && argsNode.length > expectedArgSize) {
+    const extraArgs = argsNode.slice(expectedArgSize);
+    const { from, to } = getFromAndTo(extraArgs);
     return {
       actions: [
         {
           apply(view, from, to) {
             view.dispatch({ changes: { from, to } });
           },
-          name: `Remove argument${argsNode.length > 1 ? 's' : ''}`,
+          name: `Remove ${extraArgs.length} extra argument${extraArgs.length > 1 ? 's' : ''}`,
         },
       ],
-      from: from,
-      message: 'The command should not have arguments',
+      from,
+      message: `Extra arguments, definition has ${expectedArgSize}, but ${argsNode.length} are present`,
       severity: 'error',
-      to: to,
+      to,
     };
   }
+  if ((argsNode && argsNode.length < expectedArgSize) || (!argsNode && expectedArgSize > 0)) {
+    const { from, to } = getFromAndTo(argsNode ?? [stemNode]);
+    const pluralS = expectedArgSize - (argsNode?.length ?? 0) > 1 ? 's' : '';
```

Comment:
We have a `pluralize` utility function in `utilities/text.ts` if you'd like to use that instead.